### ISSUE TITLE
class NativeSessionTokenStorage not found error

### ISSUE THREAD

**Issue by staabm** [silexphp/Silex#1349]
I am using the `FormServiceProvider` class in a app [1] in which I recently run into 

```
Fatal error: Class 'Symfony\Component\Security\Csrf\TokenStorage\NativeSessionTokenStorage' not found in /home/COMPLEX/mstaab/htdocs/tools/cronexpression/vendor/silex/silex/src/Silex/Provider/FormServiceProvider.php on line 111
```

after a `composer update`. see the related composer bugreport [2].

I am wondering why the `FormServiceProvider` relies on a class which is neither part of a declared dependency in the composer.json nor part of a suggested dependency.

ATM it seems silex just expects that a consuming project somehow provides the necessary classes for the `FormServiceProvider`.
How is this class expected to work and where should the dependency come from?

[1] https://github.com/staabm/cronschedule
[2] https://github.com/composer/composer/issues/5310

**Comment by HeahDude** [comment]
You only need the session to use csrf protection.

**Comment by staabm** [comment]
Nevertheless I would expect a more gracefull / dev friendly error message then "Class not found" with some hints on how to proceed

**Comment by HeahDude** [comment]
See http://silex.sensiolabs.org/doc/providers/form.html#registering

**Comment by ragboyjr** [comment]
@staabm, is there anything else you need or can we close this?

**Comment by staabm** [comment]
With the provided information I can fix the issue. UX wise there still is some room for improvement thou

**Comment by hkdobrev** [comment]
It seems to me @staabm was using Silex 1.x.

The report was indeed an issue as the CSRF form extension was included by default without checking for existence of the session classes. That was fixed in #1222 with the introduction of the CSRF service provider. It was released in Silex 2.0.

The last time the `FormServiceProvider.php` file had 111 lines was in Silex 1.x.

I think this should be closed now.
